### DISCUSSION TITLE
Best protocol to use with a battery-powered ESP32 plus an RF sender in the basement?

### DISCUSSION THREAD

**Discussion by sheilbronn** [merbanan/rtl_433#2441] [Q&A]
Hello,
I'd like to transfer a few bytes per hour in the 433 MhZ band from an ESP32 (Tasmota32) in the basement to a receiver on the first floor.
I'm looking for any recommendations which protocol would fit best with decoding by rtl_433 - Therefor my requirements are:

1. Simple encoding on the [Tasmota](https://tasmota.github.io/docs/) side, e.g. with [RfSend](https://tasmota.github.io/docs/RF-Protocol/)
1. Simple decoding on the rtl_433 side (i.e. a already builtin protocol ?)
1. Robust transfer (maybe longer bursts preferred?) 
1. Energy efficient (because on battery)

As an alternative, I'd be fine with an [ESPHome Remote Transmitting](https://esphome.io/components/remote_transmitter.html), too. With ESPHome: Which parameter should be used with [transmit_rc_switch_raw](https://esphome.io/components/remote_transmitter.html) ?

I would be thankful for any insights or comments and would be very willing to share experiences. Thy.

**Comment by zuckschwerdt** [comment]
Do you already have a RF sender module? Something like a CC1101 could be a good fit. It features a preamble, sync word, and after the data a CRC checksum. The PCM data needs to be Manchester coded or whitened (as in white noise). The modules can also be used in a pair as bridge (no rtl_433 needed).

If you want to use the "RfSend modules", a Superheterodyne Rx/Tx, then PWM or PPM could also be used , which is more robust and might use less power (PPM). Use 1-2 byte preamble and 1 byte (or less) of sync. A CRC is recommended but a simple SUM or XOR byte might do. You should be able to just use a Flex decoder conf for this.

Otherwise look into the Radiohead protocol if you need a ready-built decoder in rtl_433. There are libs for embedded systems available.

**Reply by letshin** [reply]
Have you tried the CC1101 with Tasmota? What's your setup please - can you share?
